File: code/Production_Applications-Chapter_15_How_Spark_Runs_on_a_Cluster.py
# Creating a SparkSession in Python
from pyspark.sql import SparkSession
spark = SparkSession.builder.master("local").appName("Word Count")\
    .config("spark.some.config.option", "some-value")\
    .getOrCreate()


# COMMAND ----------

df1 = spark.range(2, 10000000, 2)
df2 = spark.range(2, 10000000, 4)
step1 = df1.repartition(5)
step12 = df2.repartition(6)
step2 = step1.selectExpr("id * 5 as id")
step3 = step2.join(step12, ["id"])
step4 = step3.selectExpr("sum(id)")

step4.collect() # 2500000000000


# COMMAND ----------

File: code/Scala/Production_Applications-Chapter_15_How_Spark_Runs_on_a_Cluster.scala
// Creating a SparkSession in Scala
import org.apache.spark.sql.SparkSession
val spark = SparkSession.builder().appName("Databricks Spark Example")
  .config("spark.sql.warehouse.dir", "/user/hive/warehouse")
  .getOrCreate()


// COMMAND ----------

// in Scala
import org.apache.spark.SparkContext
val sc = SparkContext.getOrCreate()


// COMMAND ----------

step4.explain()


// COMMAND ----------

spark.conf.set("spark.sql.shuffle.partitions", 50)


// COMMAND ----------

